app/models/person.rb

```diff
@@ -127,6 +131,21 @@ class Person < ApplicationRecord
       .where('given_name like :pattern OR family_name like :pattern OR display_name like :pattern
         OR emails.address like :pattern', pattern: pattern)
   }
+  scope :has_listings, ->(community) do
+    joins("INNER JOIN `listings` ON `listings`.`author_id` = `people`.`id` AND `listings`.`community_id` = #{community.id} AND `listings`.`deleted` = 0").distinct
+  end
+  scope :has_no_listings, ->(community) do
+    joins("LEFT OUTER JOIN `listings` ON `listings`.`author_id` = `people`.`id` AND `listings`.`community_id` = #{community.id} AND `listings`.`deleted` = 0")
+    .where(listings: {author_id: nil}).distinct
```

Comment:
If this scope is used directly on `people` instead of through `community.members` it results in full scan of `people` table, which is obviously very bad. I suggest that we either have a comment here warning to always use this through `community.members` or adding condition:

```
.where(listings: {author_id: nil}, people: {community_id: community.id})
```

Right now, this scope is used only through `community.members`, so it's not an immediate issue, but I can easily see this being overlooked.